unittest.main stub does not accept None for module argument

The stub file for unittest appears to only allow `str` for the `module` argument to `unittest.main`, but the implementation accepts `None`.

The offending line appears to be https://github.com/python/typeshed/blob/d45fc3daaf6ba64ba2d1ee97fd32537e88aa9e7f/stdlib/3/unittest/__init__.pyi#L310

### Environment

cdunklau@hobbes test> venv/bin/python --version
Python 3.6.5
cdunklau@hobbes test> venv/bin/mypy --version
mypy 0.620

### Reproduction

File foo.py:

```python
import unittest

unittest.main(module=None, argv=['unittest', '-v', 'test_stuff'])
```

File test_stuff.py:

```python
import unittest

class DemoTestCase(unittest.TestCase):
    def test_passing(self):
        self.assertEqual(1, 1)

    def test_failing(self):
        self.assertEqual(1, 5)
```

The code runs smoothly, tests pass and fail as expected:

```
cdunklau@hobbes test> venv/bin/python foo.py
test_failing (test_stuff.DemoTestCase) ... FAIL
test_passing (test_stuff.DemoTestCase) ... ok

======================================================================
FAIL: test_failing (test_stuff.DemoTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/cdunklau/test/test_stuff.py", line 8, in test_failing
    self.assertEqual(1, 5)
AssertionError: 1 != 5

----------------------------------------------------------------------
Ran 2 tests in 0.001s

FAILED (failures=1)
```

...but mypy complains:

```
cdunklau@hobbes test> venv/bin/mypy foo.py
foo.py:3: error: Argument "module" to "main" has incompatible type "None"; expected "str"
```

### Real-world Context

It's not really clear to me if this is a supported use case for unittest, but it's the way that I figured it out. My program is bundled with PyInstaller, and I added a routine to the program itself to run the test suite so I can be confident that the built application will function as well as the installed-in-a-virtualenv version.

Since my program gets all bundled up in a zip, in a way I haven't figured out yet, the usual unittest discovery routine doesn't pick up on my test modules. I solved this by directly specifying the test modules to run in the `argv` argument as shown. This doesn't resolve the modules properly when i use the default `module='__main__'` argument to `unittest.main`, the only way I've been able to get it to run properly is by specifying `module=None`.

--- a/stdlib/2/unittest.pyi
+++ b/stdlib/2/unittest.pyi
@@ -226,7 +226,7 @@ def skip(reason: Union[str, unicode]) -> Any: ...
 class TestProgram:
     result = ...  # type: TestResult
 
-def main(module: str = ..., defaultTest: Optional[str] = ...,
+def main(module: Union[None, Text, types.ModuleType] = ..., defaultTest: Optional[str] = ...,
          argv: Optional[Sequence[str]] = ...,
          testRunner: Union[Type[TextTestRunner], TextTestRunner, None] = ...,
          testLoader: TestLoader = ..., exit: bool = ..., verbosity: int = ...,

--- a/stdlib/3/unittest/__init__.pyi
+++ b/stdlib/3/unittest/__init__.pyi
@@ -307,7 +307,7 @@ class TextTestRunner(TestRunner):
 class TestProgram:
     result = ...  # type: TestResult
 
-def main(module: str = ...,
+def main(module: Union[None, str, ModuleType] = ...,
          defaultTest: Union[str, Iterable[str], None] = ...,
          argv: Optional[List[str]] = ...,
          testRunner: Union[Type[TestRunner], TestRunner, None] = ...,

